### PR TITLE
Add support for left-justified padding in formatted prints

### PR DESCRIPTION
**Pull request by pieter3d**
This is an implementation for Issue #2101

### CODE REVIEW

**Comment by pieter3d** [comment]
The build failure is complaining about missing "pieter3d" in CONTRIBUTORS, but I added my full name instead like everyone else has. Recommended resolution to this?

**Comment by wsnyder** [comment]
Put your "real" name into ~/.gitconfig, and also use the same name in github.com, as github doesn't see your .gitconfig it uses that name. I'll update the failure to clarify this.

File: include/verilated.cpp [hunk 1]
```diff
@@ -643,7 +643,7 @@ void _vl_vsformat(std::string& output, const char* formatp, va_list ap) VL_MT_SA
                 widthSet = true;
                 width = width*10 + (fmt - '0');
                 break;
-            case '.':
+            case '.': case '-':
```
Comment (wsnyder) [inline include/verilated.cpp]: case '.':  // FALLTHRU
            case '-':

**Comment by wsnyder** [comment]
Sorry, realized you did change the runtime code, so that's probably ok.  Just need tests then, see e.g. test_regress/t/t_display*

**Comment by pieter3d** [comment]
Tests added.

File: src/V3EmitC.cpp [hunk 2]
```diff
@@ -1799,7 +1799,7 @@ void EmitCStmts::displayNode(AstNode* nodep, AstScopeName* scopenamep,
             switch (tolower(pos[0])) {
             case '0': case '1': case '2': case '3': case '4':
             case '5': case '6': case '7': case '8': case '9':
-            case '.':
+            case '.': case '-':
```
Comment (wsnyder) [inline src/V3EmitC.cpp]: Need // FALLTHRU here too

**Comment by wsnyder** [comment]
Please apply the attached, and you can remove the new test. Then running t_display_noopt.pl will get different results from t_display.pl, so there's something needing a tweak.

[issue-2111.patch.txt](https://github.com/verilator/verilator/files/4062527/issue-2111.patch.txt)

**Comment by pieter3d** [comment]
Found quite a few missing cases from that, nice catch. Added tests for those.

**Comment by wsnyder** [comment]
Merged; thanks for your efforts, much appreciated.

**Pull request closed without merging**